meadow/html/index.html

```diff
@@ -53,1134 +67,1426 @@
+            mode: 'text/x-haskell'
+          });
+          codeArea.setSize('100%', '100%');
+        </script>
+        <div id="actionButtons">
+          <div class="row1">
+            <button id="b2c" onclick="Haste.b2c()">-&gt; Blockly to Code</button>
+            <button id="c2b" onclick="Haste.c2b()">Code to Blockly &lt;-</button>
+            <button id="clear" onclick="Haste.clear_workspace()">Clear</button>
+            <button id="execute" onclick="Haste.execute()">Execute</button>
+          </div>
+          <div class="row2">
+            <button
+              id="embedding"
+              onclick="ifr=document.getElementById('iframe'); if (ifr == null) {var ifr=document.createElement('iframe');ifr.setAttribute('id', 'iframe');document.body.appendChild(ifr);ifr.setAttribute('src', 'editor/index.html');} else { ifr.style.display = 'block'; }"
+            >
+              Use Haskell embedding editor (Fay)
+            </button>
+          </div>
+        </div>
+
+        <br /><br />
+
+        <div class="group">
+          <h3 style="display:inline;" >Current block:</h3>
+	  <input style="display:inline;" type="number" id="currBlock" value="0" />
+<!--      <span class="highlight"></span>
+          <span class="bar"></span> -->
+        </div>
+
+        <div><h3>Contract state:</h3></div>
+        <div class="statusDiv" id="contractStateDiv"></div>
+        <script>
+          contractState = CodeMirror(document.getElementById('contractStateDiv'), {
+            value: '([],[])',
+            mode: 'text/x-haskell',
+            lineWrapping: true
+          });
+          contractState.setSize('100%', '100%');
+        </script>
+        <div><h3>Input:</h3></div>
+        <div class="statusDiv" id="contractInputDiv"></div>
+        <script>
+          contractInput = CodeMirror(document.getElementById('contractInputDiv'), {
+            value: '([],[],[],[])',
+            mode: 'text/x-haskell',
+            lineWrapping: true
+          });
+          contractInput.setSize('100%', '100%');
+        </script>
+        <div id="tabs">
+          <ul>
+            <li><a draggable="false" href="#tabs-1">Smart interface</a></li>
+            <li><a draggable="false" href="#tabs-2">Manual interface</a></li>
+          </ul>
+          <div id="tabs-1">
+            <table style="width:100%">
+              <thead>
+                <tr>
+                  <th width="100%"><b>Potential actions</b></th>
+                  <th align="center" valign="middle">
+                    <button onclick="Haste.refreshActions()">&nbsp;&nbsp;&nbsp;Refresh&nbsp;&nbsp;&nbsp;</button>
+                  </th>
+                </tr>
+              </thead>
+              <tbody id="actions">
+                <tr>
+                  <td align="center" valign="middle">Loading...</td>
+                  <td></td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+          <div id="tabs-2">
```

Comment:
It seems that the Manual interface now has lots of white space. In particular, it seems the controls get spread-out horizontally as you resize. Can we make it more compact? Maybe aligned to the left?

Reply:
That is correct, I made it so that it divide automatically the space for each controls. However, I suspect that you are using a large screen such as an iMac 27". Which obviously would make the buttons look awful. I'll fix that.